src/main/java/MorseCodeConverter.java

```diff
@@ -1,4 +1,58 @@
 public class MorseCodeConverter
 {
+    private final MorseCode morseCode;
 
+    public MorseCodeConverter(MorseCode morseCode)
+    {
+        this.morseCode = morseCode;
+    }
+
+    public String toMorseCode(String englishExpression)
+    {
+        englishExpression = englishExpression.toUpperCase();
+
+        StringBuilder morseExpression = new StringBuilder();
+
+        String[] englishWords = englishExpression.split(" ");
+
+        for (String word : englishWords)
+        {
+            String[] englishLetters = word.split("");
+
+            boolean letterTranslated = false;
+            for (String letter : englishLetters)
+            {
+                if (!letter.equals(""))
```

Comment:
I'm not sure it's possible to get "" with split

Reply:
This solved something that kept causing the translator to crash where extra spaces caused the letter to be processed but there not being a 0th index to work with